lib/puppet/type/concat_file.rb

```diff
@@ -58,68 +54,98 @@ def exists?
   end
 
   newparam(:order) do
-    desc "Controls the ordering of fragments. Can be set to alphabetical or numeric."
-    defaultto 'numeric'
+    desc "Controls the ordering of fragments. Can be set to :alpha or :numeric."
```

Comment:
I took another look at this and `concat_fragment` already *has* to know what the target will be, it can not be declared otherwise, see:

https://github.com/puppetlabs/puppetlabs-concat/blob/master/lib/puppet/type/concat_fragment.rb#L45-L49